.bazelrc

```diff
@@ -1,10 +1,12 @@
 # Require c++17 for c++ files.
 build --cxxopt='-std=c++17'
-# Disable warnings building zlib.
-build --copt='-Wno-deprecated-non-prototype'
+build --copt='-Wno-sign-compare'
 
 # Disable Bzlmod, a new external dependency management system on by default
 # since Bazel 7.0.0. This breaks compilation of protobuf by pulling in an older
 # version than what is specified in WORKSPACE.
 # TODO(b/319726507): Remove this when we have a better fix or move to Bzlmod.
 build --noenable_bzlmod
+
+# Load user-specific configuration, if any.
+try-import %workspace%/user.bazelrc
```

Comment:
I'm a little surprised / sad that bazel lists this as a "best practice" [link](https://bazel.build/configure/best-practices#bazelrc-file) -- user-specific configs are non-hermetic by definition and lead to poor code health in the long term. They cause exactly the "this runs and builds on my machine" problems that bazel explicitly set out to eliminate (and has been successful at eliminating internally as blaze).

I'm ok to submit as is, but @hzeller , mostly curious, what special build config are you using?

Reply:
Typically for
  * Adding things like warnings. I like to have a more tighter set of warnings in my local workspace than what necessarily should ship; `-Wall -Wextra -Werror` kindof thing. Or making sure to not accidentally depend on expensive language features `-fno-exception -fno-rtti`). Great for development, but it shouldn't unnecessarily break someone getting a project and buiding things (because the toolchain is not fixed with bazel by default:  the toolchain is whatever is on the system. (Yes, with `blaze`, everything is working. But that is because that is actually fully hermetic, as everything from toolchain to any library to depend on is built from head. `bazel` is much more leaky)
  * Adding configs for running asan, coverage or other diagnostics with particular settings which might make sense for local development but not shipping
  * Working around non-hermetic issues in the code-base. Here, it links `-lelf`, which is outside the bazel build WORKSPACE and needed a particular `--linkopt=-Lsomething` on my machine to make sure it finds the correct one.

Given the user the option to change that should always be there and yes I indeed think it is a good practice to provide a break-glass escape hatch. For a hermetic build, that is what the fresh checkout from head is.